underdocs-renderer/src/main/kotlin/underdocs/renderer/output/html/render/page/PageRenderer.kt

```diff
@@ -67,10 +67,16 @@ class PageRenderer(
                     .with(
                         UnescapedText(
                             """
-                                const isNightModePreferred = () => window.matchMedia?.('(prefers-color-scheme: dark)').matches;
-
+                                const getPreferredTheme = () => {
+                                  if (localStorage.getItem('underdocs_theme') === null) {
+                                    return window.matchMedia?.('(prefers-color-scheme: dark)').matches ? 'night' : 'day';
+                                  }
+                                
+                                  return localStorage.getItem('underdocs_theme');
+                                }
+                                
                                 (function setPreferredTheme() {
-                                  if (isNightModePreferred()) {
+                                  if (getPreferredTheme() === 'night') {
```

Comment:
There are three string literals being used all over the place: `underdocs_theme`, `day` and `night`. I think, it'd be a good idea to extract these literals into named constants (thus, for example, no typos would occur). For example:

~~~~JavaScript
const UnderdocsTheme = Object.freeze({
  DAY: 'day',
  NIGHT: 'night',
  KEY: 'underdocs_theme'
})
~~~~

Above, I used `Object.freeze` to ensure that no changes can be made to the created object (thus, it's really constant).

Then, you can write

~~~~JavaScript
if (getPreferredTheme() === UnderdocsTheme.NIGHT) {
}
~~~~

Reply:
Good idea!